src/SDKs/Azure.Base/data-plane/Azure.Base.Tests/HttpClientTransportTests.cs

```diff
@@ -305,5 +305,86 @@ public async Task RequestIdCanBeOverriden()
             var response =  await ExecuteRequest(request, transport);
             Assert.AreEqual(request.RequestId, response.RequestId);
         }
+
```

Comment:
If the intent is that this should never be writable, should we consider testing that to be sure that it isn't inadvertently changed?

Reply:
It's less of a requirement and more that we don't care about writing. I can add a test but we don't rely on writability.

Reply:
Added tests.